File: android/src/main/kotlin/com/ke/flutterqrscanner/QRScanActivity.java
package com.ke.flutterqrscanner;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.LinearLayout;

import com.google.zxing.Result;

import me.dm7.barcodescanner.zxing.ZXingScannerView;

public class QRScanActivity extends Activity implements ZXingScannerView.ResultHandler {
    static final String EXTRA_SCAN_RESULT = "EXTRA_SCAN_RESULT";



    private ZXingScannerView mScannerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_qracan);


        mScannerView = findViewById(R.id.scanner_view);

        LinearLayout linearLayout = findViewById(R.id.ll_title);
        linearLayout.setPadding(0, getStatusBarHeight(), 0, 0);

    }


    protected int getStatusBarHeight() {
        int result = 0;
        int resourceId = getResources().getIdentifier("status_bar_height", "dimen", "android");
        if (resourceId > 0) {
            result = getResources().getDimensionPixelSize(resourceId);
        }
        return result;
    }

    @Override
    public void onResume() {
        super.onResume();
        mScannerView.setResultHandler(this); // Register ourselves as a handler for scan results.
        mScannerView.startCamera();          // Start camera on resume
    }

    @Override
    public void onPause() {
        super.onPause();
        mScannerView.stopCamera();           // Stop camera on pause
    }

    @Override
    public void handleResult(Result rawResult) {
        returnResult(rawResult.getText());



    }


    private void returnResult(String result){
        Intent intent = new Intent();
        intent.putExtra(EXTRA_SCAN_RESULT, result);
        setResult(RESULT_OK, intent);
        finish();

    }

    public void back(View view) {
        onBackPressed();
    }

    public void toggleFlash(View view) {
        view.setSelected(!view.isSelected());
        boolean open = view.isSelected();
        mScannerView.setFlash(open);
    }

    public void myQRCode(View view) {
        returnResult("my_qr_code");
    }

    public void manualEnter(View view) {
        returnResult("manual_enter");
    }
}

File: android/src/main/kotlin/com/ke/flutterqrscanner/QRScanActivity.kt
package com.ke.flutterqrscanner

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.LinearLayout

import com.google.zxing.Result

import me.dm7.barcodescanner.zxing.ZXingScannerView

class QRScanActivity : Activity(), ZXingScannerView.ResultHandler {


    private var mScannerView: ZXingScannerView? = null


    private val statusBarHeight: Int
        get() {
            var result = 0
            val resourceId = resources.getIdentifier("status_bar_height", "dimen", "android")
            if (resourceId > 0) {
                result = resources.getDimensionPixelSize(resourceId)
            }
            return result
        }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_qracan)


        mScannerView = findViewById(R.id.scanner_view)

        val linearLayout = findViewById<LinearLayout>(R.id.ll_title)
        linearLayout.setPadding(0, statusBarHeight, 0, 0)

    }

    public override fun onResume() {
        super.onResume()
        mScannerView!!.setResultHandler(this) // Register ourselves as a handler for scan results.
        mScannerView!!.startCamera()          // Start camera on resume
    }

    public override fun onPause() {
        super.onPause()
        mScannerView!!.stopCamera()           // Stop camera on pause
    }

    override fun handleResult(rawResult: Result) {
        returnResult(rawResult.text)


    }


    private fun returnResult(result: String) {
        val intent = Intent()
        intent.putExtra(EXTRA_SCAN_RESULT, result)
        setResult(Activity.RESULT_OK, intent)
        finish()

    }

    fun back(view: View) {
        onBackPressed()
    }

    fun toggleFlash(view: View) {
        view.isSelected = !view.isSelected
        val open = view.isSelected
        mScannerView!!.flash = open
    }

    fun myQRCode(view: View) {
        returnResult("my_qr_code")
    }

    fun manualEnter(view: View) {
        returnResult("manual_input")
    }

    companion object {
        internal const val EXTRA_SCAN_RESULT = "EXTRA_SCAN_RESULT"
    }
}

